### PR TITLE
Update config for Gilded Rose exercise

### PR DESCRIPTION
**Pull request by fpsvogel**
Adds to `config.json` for the new Gilded Rose exercise:

- Contributors list
  - Reviewers of #1586 plus BethanyG, who gave extensive feedback in the [forum thread].
- Minimum Ruby version of 3.1
  - The tests use omitted values in hash literals, a Ruby 3.1 feature.
- ~~In the root `config.json`, removes the prerequisites for Gilded Rose. This makes the exercise more easily accessible to Ruby devs who aren't already on Exercism. (If they came for Gilded Rose, did the "Hello world" exercise, and then saw that they still needed to other exercises first, they might give up.)~~
  - _EDIT: Removed, see blow._

I also wanted to specify an `icon`, but [the existing icons](https://exercism.org/docs/building/tracks/icons#h-exercise-icons) are specific to other exercises, and googling didn't turn up any suitable free icon with a permissive license. I'm not an artist, so I'll leave off the icon for someone else to contribute.

[forum thread]: https://forum.exercism.org/t/add-the-gilded-rose-kata/7758

### CODE REVIEW

**Review by kotp — APPROVED** [review]
I think the removal of the prerequisites are fine.  We can see how that works out.

**Comment by iHiD** [comment]
> In the root config.json, removes the prerequisites for Gilded Rose. This makes the exercise more easily accessible to Ruby devs who aren't already on Exercism. (If they came for Gilded Rose, did the "Hello world" exercise, and then saw that they still needed to other exercises first, they might give up.)

I don't love this change. The same argument could be made for pretty much any exercise. I don't think this is a special exception. If someone doesn't want to do the prereqs we have a UI-level option for that, which is to switch to Practice Mode. Instead, we'll get newbies to Ruby trying to do it as it gets unlocked straight away. So I think this change should be reverted.

**Comment by kotp** [comment]
Do we need to revert this or just not accept it?  I think we may not need to revert it, it does not appear to have been accepted yet (I did not merge it only approved it pending other opinions.)

**Comment by kotp** [comment]
Also, since this is active, we can force push without fed0cd5adebaa745d9460e7dc2bf847be7657894 being included.

**Comment by iHiD** [comment]
Ah, I thought it was merged. Yeah, just leave that bit out please :)

File: config.json
Comment (kotp) [inline config.json:1606]: As @iHiD commented, let's leave the prerequisites in.  The student has a way to get around having to do the exercises for the prerequisites, and it hopefully will avoid folks trying to refactor without having a good (enough) understanding of the prerequisites.

**Review by kotp — CHANGES_REQUESTED** [review]

File: config.json
Comment (fpsvogel) [reply config.json]: Done!

**Review by kotp — APPROVED** [review]

**Pull request merged**
